Assignment2-Python/deletemiddlenode.py

```diff
@@ -0,0 +1,51 @@
+#2.3 Delete Middle Node (p. 93)
+class Node:
+    def __init__(self, data=0, next=None):
+        self.data = data
+        self.next = next
+
+def delete_node(node):
+    if node is None or node.next is None:
```

Comment:
Be careful using `is` over `==` in Python. `is` is used for [identity equality](https://www.blog.pythonlibrary.org/2017/02/28/python-101-equality-vs-identity/#:~:text=Hopefully%20by%20now%20you%20understand,object%20you%20are%20referring%20to.). Also, because `None` is already falsey in Python, you don't need to explicitly check against it here.
```suggestion
    if not node or not node.next:
```